### DISCUSSION TITLE
Want to understand why client component wont Fast-Refresh after syntax error, not happening with server side components.

### DISCUSSION THREAD

**Discussion by jphouminh71** [vercel/next.js#58227] [Help]
### Summary

So I've been developing an app using NextJS and I am enjoying the experience so far but am really stumped on how this Fast-Refreshing functionality works in regards to Client-Components. I have worked with nodemon in the past so I understand the general concept where when you update files in the application it will automatically spin up the new changes in the browser.

So the strange thing that I can't completely figure out is when im using NextJS's fast-refreshing and am working on some kind of client component.

Am running environment with NextJS 14.0.1, Google Chrome, Node 20.0.9, VSCode. Running application with npm run dev

So as expected, when working on some component I am able to see the live changes I am making on the development website and things are fine, but what is happening is that when I am working on a component that marked with 'use client' and I make a syntax error, the development website shows up saying syntax error, so I go ahead and fix it and save the changes and the development website recompiles and shows up on the screen. But now any subsequent changes I make to the code I am not able to see on the development site until I manually refresh the chrome browser then things are all happy and fine again. This behavior doesn't happen when working with server components.

TLDR: When working with client components and if I make a syntax error and fix it, I wont be able to see the live changes to my page until I manually refresh the chrome browser, doesn't happen with server components.

Spent all night reading about NextJS, came upon the caching topic, but not sure if its directly related to the issue I am facing with just developing

Here is a link to a repo that can be pulled down to recreate the issue: https://github.com/jphouminh71/WebDev/tree/master/NextJS/fast-reload-issues/app

**Comment by aligajani** [comment]
Puzzling me here too.
